### DISCUSSION TITLE
[Camera Support]: Supporting Annke DN81R ONVIF PTZ

### DISCUSSION THREAD

**Discussion by skotzkoj** [blakeblackshear/frigate#10568] [Camera Support]
### Describe the problem you are having

I'm trying to setup my Annke DN81R to support PTZ. I know it's capable because homeassistant picks up the NVR cameras and can send commands to drive the PTZ enabled cameras. I cannot find a way to specify which camera from my NVM to send the ONVIF PTZ commands to. Does Frigate support the onvif media profiletokens? 

### Version

0.13.2-6476F8A

### Frigate config file

```yaml
N/A
```

### Relevant log output

```shell
N/A
```

### FFprobe output from your camera

```shell
N/A
```

### Operating system

Debian

### Install method

Docker CLI

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

Annke DN81R

**Comment by hawkeye217** [comment]
There have been recent updates to the onvif code in Frigate (https://github.com/blakeblackshear/frigate/pull/10012 and https://github.com/blakeblackshear/frigate/pull/10350) which will likely fix your issue. These have been merged and will be available in the next release.

**Comment by DerPicknicker** [comment]
Any updates on this? Does the camera work now fine @skotzkoj ?

**Reply by hawkeye217** [reply]
The fixes committed above will be released with Frigate 0.14. The current stable version of Frigate is 0.13.2.

**Reply by DerPicknicker** [reply]
@hawkeye217 .. I am quite new to frigate. Does that mean that ALL PTZ cameras from Annke and so one are supported? Or do I have to choose one of the supported one from the docs? 

Because the Annke ones are quite cheap and seems like a good choice... The Autotracking-Feature is a must for me.

I want that frigate is the NVR and directly communicating with the cameras without NVR from the Camera manufacturer.

**Reply by NickM-27** [reply]
> The Autotracking-Feature is a must for me.

then annke is probably not a brand you should be looking at, as their onvif implementation has known bugs that stop autotracking from working

**Reply by DerPicknicker** [reply]
> has known bugs that stop autotracking from working

What would you recommend? It should be used outdoor. So would be good if it has optical zoom and a crisp image.

**Reply by hawkeye217** [reply]
The camera features required for frigate's autotracking are not often found on inexpensive/low-end cameras. For outdoor cameras, the cheapest I've found personally is the Dahua SD1A404XB-GNR. Users have also reported success with some Amcrest models.

More and more cameras these days also have onboard autotracking built into the firmware. Because frigate's implementation of autotracking uses ONVIF, firmware-based solutions usually outperform frigate in that regard.

**Reply by DerPicknicker** [reply]
@hawkeye217 .. Thank you for the clarification - then I will buy a cheap one and use only onvif for getting audio and video.
